src/platform/android/DnssdImpl.cpp

```diff
@@ -223,7 +223,7 @@ void HandleResolve(jstring instanceName, jstring serviceType, jstring address, j
         callback(reinterpret_cast<void *>(contextHandle), service, Span<Inet::IPAddress>(), error);
     };
 
-    VerifyOrReturn(address != nullptr && port != 0, dispatch(CHIP_ERROR_UNKNOWN_RESOURCE_ID));
+//     VerifyOrReturn(address != nullptr && port != 0, dispatch(CHIP_ERROR_UNKNOWN_RESOURCE_ID));
```

Comment:
What's wrong with validating address and port? In which circumstances can they be 0s?

Reply:
I don't understand the lower layers enough but when I tried to "PROVISION CHIP DEVICE WITH WIFI" or "LIGHT ON/OFF & LEVEL CLUSTER" the code gets to that verify and exits printing this out twice in log cat:
`2022-03-03 16:05:14.326 16902-17436/com.google.chip.chiptool E/CTL: Error resolving node id: ConnectedHomeIP/src/platform/android/DnssdImpl.cpp:259: CHIP Error 0x000000A6: Unknown resource ID`

Without the verification, the android app works and I can provision the ESP32, turn the lights On and Off, read the status of the ESP32's light and its temperature.

I've done some improvement in commit https://github.com/project-chip/connectedhomeip/pull/15919/commits/668bb4657723a0fab45c1a59f5d65309812bf0ae this time keeping the verification but fixing the expression as the `VerifyOrReturn ` will act on the parameters "if expression evaluates to false". 
I thought the old `address != nullptr && port != 0` and new `!(address == nullptr || port == 0)` expression should evaluate the same but clearly they don't as it works with the new but not the old one.